backend/flindt/feedback/views.py

```diff
@@ -40,8 +40,17 @@ def get_queryset(self):
         """
         Set the queryset here so it filters on user.
         """
-        return super(FeedbackViewSet, self).get_queryset().filter(Q(sender=self.request.user) |
-                                                                  Q(recipient=self.request.user))
+        data = self.request.data
+        feedback_type = data.get('feedbackType', None)
+        if not feedback_type:
+            return None
+
+        # Filter whether the user is the sender or recipient of the feedback.
+        # To prevent that the sender can change the feedback of the recipient.
+        if feedback_type == 'receive':
+            return super(FeedbackViewSet, self).get_queryset().filter(sender=self.request.user)
+        elif feedback_type == 'give':
+            return super(FeedbackViewSet, self).get_queryset().filter(recipient=self.request.user)
 
```

Comment:
I would add an "else return BadRequest" to indicate the feedback_type is invalid

Reply:
One more thing, I'm not sure if this change actually works from a security standpoint. Sure we could change our frontend to fix it. But if the request has a parameter feedbackType, what prevents someone to just try to change the parameter manually and try to change a feedback with type 'receiver'? (Using a custom client, not the frontend).